src/data/polynomial/monic.lean

```diff
@@ -397,38 +407,42 @@ begin
   { rw degree_le_iff_coeff_zero,
     intros m hm,
     rw degree_lt_iff_coeff_zero at hm,
-    refine h _ _,
+    refine h _,
     simpa using hm m le_rfl },
 end
 
-lemma nat_degree_smul_of_non_zero_divisor {S : Type*} [monoid S] [distrib_mul_action S R]
-  (p : polynomial R) (k : S) (h : ∀ (x : R), k • x = 0 → x = 0) :
+lemma nat_degree_smul_of_smul_regular {S : Type*} [monoid S] [distrib_mul_action S R]
+  {k : S} (p : polynomial R) (h : is_smul_regular R k) :
   (k • p).nat_degree = p.nat_degree :=
 begin
   by_cases hp : p = 0,
   { simp [hp] },
   rw [←with_bot.coe_eq_coe, ←degree_eq_nat_degree hp, ←degree_eq_nat_degree,
-      degree_smul_of_non_zero_divisor _ _ h],
+      degree_smul_of_smul_regular p h],
   contrapose! hp,
-  rw polynomial.ext_iff at hp ⊢,
-  simp only [coeff_smul, coeff_zero] at hp,
-  intro n,
-  simp [h _ (hp n)]
+  -- TODO: is this general over things with some smul class??
+  have h' : is_smul_regular (polynomial R) k,
+  { intros p q hpq,
+    rw ext_iff at hpq ⊢,
+    intro n,
+    refine h _,
+    simpa using hpq n },
```

Comment:
I don't think we have such a typeclass, but I would imagine this property is inherited transitively just like [`has_faithful_scalar`](https://leanprover-community.github.io/mathlib_docs/group_theory/group_action/defs.html#has_faithful_scalar) is.

Reply:
Is this statement true:
```lean
lemma is_smul_regular.tower [is_scalar_tower S R M] (k : S) (h : is_smul_regular R k) : is_smul_regular M k := sorry
```
because that would apply here.

Reply:
I don't think this is true as stated. `M` could have all sorts of extra elements that destroy the regularity of `k`.